src/Microsoft.AspNet.Mvc.TagHelpers/AnchorTagHelper.cs

```diff
@@ -24,9 +24,8 @@ public class AnchorTagHelper : TagHelper
         private const string RouteAttributePrefix = "asp-route-";
         private const string Href = "href";
 
-        // Protected to ensure subclasses are correctly activated. Internal for ease of use when testing.
-        [Activate]
-        protected internal IHtmlGenerator Generator { get; set; }
+        [Activate, HtmlAttributeNotBound]
```

Comment:
Not a guideline, but it's what we always do. Yes, please clean up. (A simple regex can fix it.)